internal/controllers/nft_controller.go

```diff
@@ -837,3 +840,200 @@ func (nc *NFTController) GetVinCredential(c *fiber.Ctx) error {
 	c.Set(fiber.HeaderContentType, fiber.MIMEApplicationJSON)
 	return c.Send(nft.R.Claim.Credential.JSON)
 }
+
+// GetBurnDevice godoc
+// @Description Returns the data the user must sign in order to burn the device.
+// @Tags        user-devices
+// @Param       tokenID path int true "token id"
+// @Success     200          {object} apitypes.TypedData
+// @Security    BearerAuth
+// @Router      /vehicle/:tokenID/commands/burn [get]
+func (udc *UserDevicesController) GetBurnDevice(c *fiber.Ctx) error {
+	tis := c.Params("tokenID")
+	ti, ok := new(big.Int).SetString(tis, 10)
+	if !ok {
+		return fiber.NewError(fiber.StatusBadRequest, fmt.Sprintf("Couldn't parse token id %q.", tis))
+	}
+	tid := types.NewNullDecimal(new(decimal.Big).SetBigMantScale(ti, 0))
+
+	client := registry.Client{
+		Producer:     udc.producer,
+		RequestTopic: "topic.transaction.request.send",
+		Contract: registry.Contract{
+			ChainID: big.NewInt(udc.Settings.DIMORegistryChainID),
+			Address: common.HexToAddress(udc.Settings.DIMORegistryAddr),
+			Name:    "DIMO",
+			Version: "1",
+		},
+	}
+
+	tx, err := udc.DBS().Reader.BeginTx(c.Context(), nil)
+	if err != nil {
+		return fiber.NewError(fiber.StatusBadRequest, err.Error())
+	}
+	defer tx.Rollback() //nolint
+
+	vehicleNFT, err := models.VehicleNFTS(
+		models.VehicleNFTWhere.TokenID.EQ(tid),
+		qm.Load(qm.Rels(models.VehicleNFTRels.UserDevice)),
+		qm.Load(qm.Rels(models.VehicleNFTRels.MintRequest)),
+		qm.Load(qm.Rels(models.VehicleNFTRels.VehicleTokenAftermarketDevice)),
+		qm.Load(qm.Rels(models.VehicleNFTRels.VehicleTokenSyntheticDevice)),
+		qm.Load(qm.Rels(models.VehicleNFTRels.UserDevice, models.UserDeviceRels.UserDeviceAPIIntegrations)),
+	).One(c.Context(), tx)
+	if err != nil {
+		return fiber.NewError(fiber.StatusBadRequest, "failed to find device by token id: %s", tis)
```

Comment:
If the NFT does not exist, then we should return a 404 since the object is not found. Otherwise, we can do a 500 and log the error. I think you can check if the errors.Is(sql.NoRow) to determine if it should be 404 or 500